src/realm/sync/noinst/client_reset.cpp

```diff
@@ -655,7 +652,7 @@ LocalVersionIDs perform_client_reset_diff(DBRef db_local, DBRef db_remote, sync:
         // partially recovered, but interrupted may continue sync the next time it is
         // opened with only partially recovered state while having lost the history of any
         // offline modifications.
-        history_local->set_client_file_ident_in_wt(wt_local->get_version(), client_file_ident);
+        history_local->set_client_file_ident_in_wt(wt_local->get_version(), remote_ident);
```

Comment:
Doesn't the server complain that we BIND needing a file ident which server provides, and then we IDENT with a different file ident?

Reply:
One curiosity, why can't we set the file ident at line 632? (would've helped if the comment at line 631 explained why)

Reply:
I think we don't set the ident on 632 because if the recovery is interrupted at some point before finishing, we still want the server to send a client reset due to having the old ident. In FLX recovery there is a series of commits and we only want to use a new ident once everything has been successfully committed.

Reply:
The server doesn't seem to complain about using the other IDENT. But that change isn't necessary here, and I think I'll restore the original just in case it does in fact matter to the server somehow.